internal/gatewayapi/route.go

```diff
@@ -112,7 +112,7 @@ func (t *Translator) processHTTPRouteParentRefs(httpRoute *HTTPRouteContext, res
 		// any conditions that come out of it have to go on each RouteParentStatus,
 		// not on the Route as a whole.
 		routeRoutes, err := t.processHTTPRouteRules(httpRoute, parentRef, resources)
-		if err != nil {
+		if err != nil && !parentRef.HasCondition(httpRoute, gwapiv1.RouteConditionAccepted, metav1.ConditionFalse) {
```

Comment:
curious why we need to change the status generation logic

Reply:
To clarify why the traffic cannot be routed, previously, the route was accepted, and the traffic was considered routable before this change.

Reply:
trying to better understand why do we need an additional check here, isnt `if err != nil` enough ?

Reply:
At this stage, it is already possible that a route is not applicable due to other reasons. This check attempts to retain the original reason, as it may be more specific. However, if that is the case, no traffic would be routed regardless to the settings validations errors.

Reply:
can you share an example of what the status looks like with and w/o this change ?

Reply:
hmm actually you are right this does not seem to happen.